Stop autoreleasing WKWebView and WKContentView in WebKit::PageClientImplIOS

ddkilzer opened the pull request:
#### 3fe8f3b95be547534fd945e5b0a9786f09ef237a
<pre>
Stop autoreleasing WKWebView and WKContentView in WebKit::PageClientImplIOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=258051">https://bugs.webkit.org/show_bug.cgi?id=258051</a>
&lt;rdar://110740310&gt;

Reviewed by Alex Christensen.

Replace m_contentView with contentView() (and m_webView with webView())
in PageClientImplIOS.mm to prevent those objects from being
autoreleased.

* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
(WebKit::PageClientImpl::contentView): Add.
- Add convenience method to call WeakObjCPtr&lt;WKContentView&gt;::get() to
  return a RetainPtr&lt;WKContentView&gt;, which avoids autoreleasing the
  object.

* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::createDrawingAreaProxy):
(WebKit::PageClientImpl::requestScroll):
(WebKit::PageClientImpl::viewScrollPosition):
(WebKit::PageClientImpl::viewSize):
(WebKit::PageClientImpl::isViewWindowActive):
(WebKit::PageClientImpl::isViewFocused):
(WebKit::PageClientImpl::isViewInWindow):
(WebKit::PageClientImpl::processDidExit):
(WebKit::PageClientImpl::processWillSwap):
(WebKit::PageClientImpl::didRelaunchProcess):
(WebKit::PageClientImpl::didCreateContextInWebProcessForVisibilityPropagation):
(WebKit::PageClientImpl::didCreateContextInGPUProcessForVisibilityPropagation):
(WebKit::PageClientImpl::gpuProcessDidExit):
(WebKit::PageClientImpl::didNotHandleTapAsClick):
(WebKit::PageClientImpl::didHandleTapAsHover):
(WebKit::PageClientImpl::didCompleteSyntheticClick):
(WebKit::PageClientImpl::didStartProvisionalLoadForMainFrame):
(WebKit::PageClientImpl::didCommitLoadForMainFrame):
(WebKit::PageClientImpl::disableDoubleTapGesturesDuringTapIfNecessary):
(WebKit::PageClientImpl::handleSmartMagnificationInformationForPotentialTap):
(WebKit::PageClientImpl::documentRect const):
(WebKit::PageClientImpl::registerEditCommand):
(WebKit::PageClientImpl::clearAllEditCommands):
(WebKit::PageClientImpl::canUndoRedo):
(WebKit::PageClientImpl::executeUndoRedo):
(WebKit::PageClientImpl::accessibilityWebProcessTokenReceived):
(WebKit::PageClientImpl::interpretKeyEvent):
(WebKit::PageClientImpl::positionInformationDidChange):
(WebKit::PageClientImpl::selectionDidChange):
(WebKit::PageClientImpl::assistiveTechnologyMakeFirstResponder):
(WebKit::PageClientImpl::screenToRootView):
(WebKit::PageClientImpl::rootViewToScreen):
(WebKit::PageClientImpl::accessibilityScreenToRootView):
(WebKit::PageClientImpl::rootViewToAccessibilityScreen):
(WebKit::PageClientImpl::doneWithKeyEvent):
(WebKit::PageClientImpl::doneWithTouchEvent):
(WebKit::PageClientImpl::doneDeferringTouchStart):
(WebKit::PageClientImpl::doneDeferringTouchMove):
(WebKit::PageClientImpl::doneDeferringTouchEnd):
(WebKit::PageClientImpl::requestTextRecognition):
(WebKit::PageClientImpl::dataOwnerForPasteboard const):
(WebKit::PageClientImpl::setTextIndicator):
(WebKit::PageClientImpl::clearTextIndicator):
(WebKit::PageClientImpl::setTextIndicatorAnimationProgress):
(WebKit::PageClientImpl::makeViewBlank):
(WebKit::PageClientImpl::setRemoteLayerTreeRootNode):
(WebKit::PageClientImpl::commitPotentialTapFailed):
(WebKit::PageClientImpl::didGetTapHighlightGeometries):
(WebKit::PageClientImpl::didCommitLayerTree):
(WebKit::PageClientImpl::layerTreeCommitComplete):
(WebKit::PageClientImpl::elementDidFocus):
(WebKit::PageClientImpl::updateInputContextAfterBlurringAndRefocusingElement):
(WebKit::PageClientImpl::isFocusingElement):
(WebKit::PageClientImpl::elementDidBlur):
(WebKit::PageClientImpl::focusedElementDidChangeInputMode):
(WebKit::PageClientImpl::didUpdateEditorState):
(WebKit::PageClientImpl::showPlaybackTargetPicker):
(WebKit::PageClientImpl::handleRunOpenPanel):
(WebKit::PageClientImpl::showShareSheet):
(WebKit::PageClientImpl::showContactPicker):
(WebKit::PageClientImpl::showInspectorHighlight):
(WebKit::PageClientImpl::hideInspectorHighlight):
(WebKit::PageClientImpl::showInspectorIndication):
(WebKit::PageClientImpl::hideInspectorIndication):
(WebKit::PageClientImpl::enableInspectorNodeSearch):
(WebKit::PageClientImpl::disableInspectorNodeSearch):
(WebKit::PageClientImpl::scrollingNodeScrollViewWillStartPanGesture):
(WebKit::PageClientImpl::scrollingNodeScrollViewDidScroll):
(WebKit::PageClientImpl::scrollingNodeScrollWillStartScroll):
(WebKit::PageClientImpl::scrollingNodeScrollDidEndScroll):
(WebKit::PageClientImpl::navigationGestureDidEnd):
(WebKit::PageClientImpl::userInterfaceLayoutDirection):
(WebKit::PageClientImpl::didPerformDragOperation):
(WebKit::PageClientImpl::didHandleDragStartRequest):
(WebKit::PageClientImpl::didHandleAdditionalDragItemsRequest):
(WebKit::PageClientImpl::startDrag):
(WebKit::PageClientImpl::willReceiveEditDragSnapshot):
(WebKit::PageClientImpl::didReceiveEditDragSnapshot):
(WebKit::PageClientImpl::didChangeDragCaretRect):
(WebKit::PageClientImpl::requestPasswordForQuickLookDocument):
(WebKit::PageClientImpl::requestDOMPasteAccess):
(WebKit::PageClientImpl::cancelPointersForGestureRecognizer):
(WebKit::PageClientImpl::activeTouchIdentifierForGestureRecognizer):
(WebKit::PageClientImpl::handleAutocorrectionContext):
(WebKit::PageClientImpl::showDataDetectorsUIForPositionInformation):
(WebKit::PageClientImpl::didEnterFullscreen):
(WebKit::PageClientImpl::didExitFullscreen):
(WebKit::PageClientImpl::writePromisedAttachmentToPasteboard):
(WebKit::PageClientImpl::setMouseEventPolicy):
(WebKit::PageClientImpl::showMediaControlsContextMenu):
(WebKit::PageClientImpl::handleAsynchronousCancelableScrollEvent):
(WebKit::PageClientImpl::runModalJavaScriptDialog):
(WebKit::PageClientImpl::contentViewBackgroundColor):
(WebKit::PageClientImpl::insertionPointColor):
(WebKit::PageClientImpl::requestScrollToRect):
(WebKit::PageClientImpl::sceneID):
(WebKit::PageClientImpl::beginTextRecognitionForFullscreenVideo):
(WebKit::PageClientImpl::cancelTextRecognitionForFullscreenVideo):
(WebKit::PageClientImpl::isTextRecognitionInFullscreenVideoEnabled const):
(WebKit::PageClientImpl::beginTextRecognitionForVideoInElementFullscreen):
(WebKit::PageClientImpl::cancelTextRecognitionForVideoInElementFullscreen):
(WebKit::PageClientImpl::hasResizableWindows const):

Canonical link: <a href="https://commits.webkit.org/265157@main">https://commits.webkit.org/265157@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bab8f10a99f38d96f4cc2c01c2f074fd05910d95

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9983 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10229 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/10481 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/11634 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9662 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/12216 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10181 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12602 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10138 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10918 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8448 "53 api tests failed or timed out") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12018 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8225 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16375 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/9396 "Build was cancelled. Recent messages:OS: Ventura (13.3), Xcode: 14.3") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9196 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12460 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9695 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7868 "4 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8854 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13083 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1129 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9475 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
```diff
@@ -114,31 +114,32 @@ - (BOOL)_isHostedInAnotherProcess;
 {
-    return (isViewInWindow() && ![m_webView _isBackground] && [m_webView _contentViewIsFirstResponder]) || [m_webView _isRetainingActiveFocusedState];
+    auto webView = this->webView();
+    return (webView && isViewInWindow() && ![webView _isBackground] && [webView _contentViewIsFirstResponder]) || [webView _isRetainingActiveFocusedState];
```
The webView check is redundant.

ddkilzer replied:
Thanks!  Fixed.

achristensen07 reviewed (approved):
It might be worth investigating what removing "operator ValueType *" would do.

Commit:
Stop autoreleasing WKWebView and WKContentView in WebKit::PageClientImplIOS
https://bugs.webkit.org/show_bug.cgi?id=258051
<rdar://110740310>

Reviewed by Alex Christensen.

Replace m_contentView with contentView() (and m_webView with webView())
in PageClientImplIOS.mm to prevent those objects from being
autoreleased.

* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
(WebKit::PageClientImpl::contentView): Add.
- Add convenience method to call WeakObjCPtr<WKContentView>::get() to
  return a RetainPtr<WKContentView>, which avoids autoreleasing the
  object.

* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::createDrawingAreaProxy):
(WebKit::PageClientImpl::requestScroll):
(WebKit::PageClientImpl::viewScrollPosition):
(WebKit::PageClientImpl::viewSize):
(WebKit::PageClientImpl::isViewWindowActive):
(WebKit::PageClientImpl::isViewFocused):
(WebKit::PageClientImpl::isViewInWindow):
(WebKit::PageClientImpl::processDidExit):
(WebKit::PageClientImpl::processWillSwap):
(WebKit::PageClientImpl::didRelaunchProcess):
(WebKit::PageClientImpl::didCreateContextInWebProcessForVisibilityPropagation):
(WebKit::PageClientImpl::didCreateContextInGPUProcessForVisibilityPropagation):
(WebKit::PageClientImpl::gpuProcessDidExit):
(WebKit::PageClientImpl::didNotHandleTapAsClick):
(WebKit::PageClientImpl::didHandleTapAsHover):
(WebKit::PageClientImpl::didCompleteSyntheticClick):
(WebKit::PageClientImpl::didStartProvisionalLoadForMainFrame):
(WebKit::PageClientImpl::didCommitLoadForMainFrame):
(WebKit::PageClientImpl::disableDoubleTapGesturesDuringTapIfNecessary):
(WebKit::PageClientImpl::handleSmartMagnificationInformationForPotentialTap):
(WebKit::PageClientImpl::documentRect const):
(WebKit::PageClientImpl::registerEditCommand):
(WebKit::PageClientImpl::clearAllEditCommands):
(WebKit::PageClientImpl::canUndoRedo):
(WebKit::PageClientImpl::executeUndoRedo):
(WebKit::PageClientImpl::accessibilityWebProcessTokenReceived):
(WebKit::PageClientImpl::interpretKeyEvent):
(WebKit::PageClientImpl::positionInformationDidChange):
(WebKit::PageClientImpl::selectionDidChange):
(WebKit::PageClientImpl::assistiveTechnologyMakeFirstResponder):
(WebKit::PageClientImpl::screenToRootView):
(WebKit::PageClientImpl::rootViewToScreen):
(WebKit::PageClientImpl::accessibilityScreenToRootView):
(WebKit::PageClientImpl::rootViewToAccessibilityScreen):
(WebKit::PageClientImpl::doneWithKeyEvent):
(WebKit::PageClientImpl::doneWithTouchEvent):
(WebKit::PageClientImpl::doneDeferringTouchStart):
(WebKit::PageClientImpl::doneDeferringTouchMove):
(WebKit::PageClientImpl::doneDeferringTouchEnd):
(WebKit::PageClientImpl::requestTextRecognition):
(WebKit::PageClientImpl::dataOwnerForPasteboard const):
(WebKit::PageClientImpl::setTextIndicator):
(WebKit::PageClientImpl::clearTextIndicator):
(WebKit::PageClientImpl::setTextIndicatorAnimationProgress):
(WebKit::PageClientImpl::makeViewBlank):
(WebKit::PageClientImpl::setRemoteLayerTreeRootNode):
(WebKit::PageClientImpl::commitPotentialTapFailed):
(WebKit::PageClientImpl::didGetTapHighlightGeometries):
(WebKit::PageClientImpl::didCommitLayerTree):
(WebKit::PageClientImpl::layerTreeCommitComplete):
(WebKit::PageClientImpl::elementDidFocus):
(WebKit::PageClientImpl::updateInputContextAfterBlurringAndRefocusingElement):
(WebKit::PageClientImpl::isFocusingElement):
(WebKit::PageClientImpl::elementDidBlur):
(WebKit::PageClientImpl::focusedElementDidChangeInputMode):
(WebKit::PageClientImpl::didUpdateEditorState):
(WebKit::PageClientImpl::showPlaybackTargetPicker):
(WebKit::PageClientImpl::handleRunOpenPanel):
(WebKit::PageClientImpl::showShareSheet):
(WebKit::PageClientImpl::showContactPicker):
(WebKit::PageClientImpl::showInspectorHighlight):
(WebKit::PageClientImpl::hideInspectorHighlight):
(WebKit::PageClientImpl::showInspectorIndication):
(WebKit::PageClientImpl::hideInspectorIndication):
(WebKit::PageClientImpl::enableInspectorNodeSearch):
(WebKit::PageClientImpl::disableInspectorNodeSearch):
(WebKit::PageClientImpl::scrollingNodeScrollViewWillStartPanGesture):
(WebKit::PageClientImpl::scrollingNodeScrollViewDidScroll):
(WebKit::PageClientImpl::scrollingNodeScrollWillStartScroll):
(WebKit::PageClientImpl::scrollingNodeScrollDidEndScroll):
(WebKit::PageClientImpl::navigationGestureDidEnd):
(WebKit::PageClientImpl::userInterfaceLayoutDirection):
(WebKit::PageClientImpl::didPerformDragOperation):
(WebKit::PageClientImpl::didHandleDragStartRequest):
(WebKit::PageClientImpl::didHandleAdditionalDragItemsRequest):
(WebKit::PageClientImpl::startDrag):
(WebKit::PageClientImpl::willReceiveEditDragSnapshot):
(WebKit::PageClientImpl::didReceiveEditDragSnapshot):
(WebKit::PageClientImpl::didChangeDragCaretRect):
(WebKit::PageClientImpl::requestPasswordForQuickLookDocument):
(WebKit::PageClientImpl::requestDOMPasteAccess):
(WebKit::PageClientImpl::cancelPointersForGestureRecognizer):
(WebKit::PageClientImpl::activeTouchIdentifierForGestureRecognizer):
(WebKit::PageClientImpl::handleAutocorrectionContext):
(WebKit::PageClientImpl::showDataDetectorsUIForPositionInformation):
(WebKit::PageClientImpl::didEnterFullscreen):
(WebKit::PageClientImpl::didExitFullscreen):
(WebKit::PageClientImpl::writePromisedAttachmentToPasteboard):
(WebKit::PageClientImpl::setMouseEventPolicy):
(WebKit::PageClientImpl::showMediaControlsContextMenu):
(WebKit::PageClientImpl::handleAsynchronousCancelableScrollEvent):
(WebKit::PageClientImpl::runModalJavaScriptDialog):
(WebKit::PageClientImpl::contentViewBackgroundColor):
(WebKit::PageClientImpl::insertionPointColor):
(WebKit::PageClientImpl::requestScrollToRect):
(WebKit::PageClientImpl::sceneID):
(WebKit::PageClientImpl::beginTextRecognitionForFullscreenVideo):
(WebKit::PageClientImpl::cancelTextRecognitionForFullscreenVideo):
(WebKit::PageClientImpl::isTextRecognitionInFullscreenVideoEnabled const):
(WebKit::PageClientImpl::beginTextRecognitionForVideoInElementFullscreen):
(WebKit::PageClientImpl::cancelTextRecognitionForVideoInElementFullscreen):
(WebKit::PageClientImpl::hasResizableWindows const):

Canonical link: https://commits.webkit.org/265157@main
